README.rst

```diff
@@ -9,12 +9,91 @@ mily
         :target: https://pypi.python.org/pypi/mily
 
 
-A collection of Qt building blocks for scientific GUIs.
+A collection of Qt building blocks for scientific and control GUIs in Python
 
 * Free software: 3-clause BSD license
 * Documentation: (COMING SOON!) https://NSLS-II.github.io/mily.
 
-Features
---------
+Project Goals
+-------------
 
-* TODO
+Across the light sources we have a range of Qt based tools covering
+the full range of application: from engineering screens, to analysis
+GUIs.  Each of these projects wrapping and extending the base Qt
+Widgets in very similar ways that are both a duplication of effort and
+may reduce the ease with which we can combine the projects.  The
+high-level goals of this project are:
+
+ - code re-use between projects
+ - similar look-and-feel across fleet
+ - ability to make applications with cross-cutting functionality
+ - make sure we are all talking to each other
+
+
+
+In Scope
+++++++++
+
+- "smarter" versions of core Qt widgets (ex, float-aware ``QSlider``,
+  spin-box + label, display alarm state)
+- domain-specific visualization
+- helper functions for managing Widgets / windows
+- dialog to show an exception to the user
+- auto-form generation tools
+
+Out of Scope
+++++++++++++
+
+- data reduction, transformation, or analysis code
+- control-system aware code
+- data I/O code
+- business logic
+
+Related projects
+++++++++++++++++
+
+
+`Xi-Cam 2 <https://github.com/lbl-camera/Xi-cam.gui>`__
+    An extensible platform for synchrotron data reduction,
+    visualization, and management.
+
+`pydm <https://github.com/slaclab/pydm>`__
+   PyDM is a PyQt-based framework for building user interfaces for
+   control systems
+
+`databroker-browser <https://github.com/NSLS-II/databroker-browser>`__
+    This is GUI for browsing runs that are stored in the databroker.
+
+blusky-qtui
+    Qt widgets for capturing user input to `blueksy
+    <https://github.com/nsls-ii/bluesky>`__ plans
+
+`typhon <https://github.com/pcdshub/typhon>`__
+    Automated User Interface Creation from Ophyd Devices
+
+`caproto-image-viewer <https://github.com/klauer/caproto-image-viewer>`__
+    A high-performance openGL based image display widget
+
+`xray-vision <https://github.com/Nikea/xray-vision>`__
+   Domain specific plotting and GUI widgets for X-ray science
+
+`adviewer <https://github.com/pcdshub/adviewer>`__
+   AreaDetector configuration and viewing / based on typhon, pydm, and ophyd
+
+
+Dependencies
+------------
+
+- qtpy
+- pyqtgraph (optional)
+- matplotlib (optional)
+
+Other dependencies (databroker, xarray, scikit-beam, scipy, etc) may
+be injected into widgets.
+
+- widgets that take in structured meta-data should take in documents
```

Comment:
This is the only thing I'm slightly unsure about. No problem with "data" being entered in the event model, but does this include the setup of other widgets. For instance, using the spinbox with label example, it would be nice if this could be setup with a dictionary of configuration information about a `Signal` but are we going to enforce that this is a describe document?

Reply:
I am also a bit skeptical to enforce the `event model`.

Reply:
Sorry, I think this is data/meta-data meaning getting crosswise.  I agree with @teddyrendahl that we do not need full document model from configuration-of-widgets dictionaries!

I was thinking more of "I am a widget that takes an x-y pair and then plots a time-trace of them" or https://github.com/lbl-camera/Xi-cam.gui/pull/4/files .

It _may_ be in scope to put Qt aware versions of some of the core-callbacks from blue sky here (ex, table, some of the plotting) so it can be shared by {xi-cam, databroker-browser, bluesky-ui}.

Reply:
I think we should revisit this down the road a bit.

Reply:
To elaborate, as long as we _may_ use event-model in places, as an optional dependency inside the widget, same as others listed above, we can wait on the decision of whether the library should take a position of whether widgets "should" do so where applicable.

Reply:
Also, nothing blocks the library to create a section of:
`Event Model Aware Widgets`... or something like that...